pkg/tfbridge/defaults/tokens.go

```diff
@@ -0,0 +1,928 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package defaults
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/Masterminds/semver"
+
+	b "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	md "github.com/pulumi/pulumi-terraform-bridge/v3/unstable/metadata"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/cgstrings"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+const (
+	defaultMinimumModuleSize    = 5
+	defaultMinimumSubmoduleSize = 5
+)
+
+// A function that joins a module and name into a pulumi type token.
+//
+// For example:
+//
+//	func(module, name string) (string, error) {
+//	 return fmt.Sprintf("pkgName:%s:%s", module, name), nil
+//	}
+type MakeToken func(module, name string) (string, error)
+
+// A strategy that assigns all tokens to the same module.
+//
+// For example:
+//
+//	rStrat, dStrat := TokensSingleModule("pkgName_", "index", finalize)
```

Comment:
Should this be `TokensSingleModule("pkgName_", "index", MakeStandardToken)`?

Should this use functional options pattern and default the finalize to MakeStandardToken?